### PR TITLE
Remove static content from docs/

### PR DESCRIPTION
**Pull request by cblecker**
**What this PR does / why we need it**:
Per the discussion in #46348, this is a first pass at removing as much as possible from docs/

Three key changes:
- Changed any references of `http://releases.k8s.io/HEAD/docs/devel/api-conventions.md` to `https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md`
- Removed everything except BUILD and OWNERS from docs
- Regenerated anything in docs that is currently auto-generated

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

@kubernetes/sig-docs-maintainers @chenopis @ahmetb @mikedanese @thockin

### CODE REVIEW

**Comment by thockin** [comment]
There are references to generated API docs in:

pkg/kubectl/cmd/patch.go
pkg/kubectl/cmd/replace.go

Don't forget federation/docs/api-reference/

I think these files are/were copied to the docs repo, but I don't know how
or when or by whom, or what we will do after they are removed.

**Comment by cblecker** [comment]
@thockin I left anything that's autogenerated, including the kubectl man pages, and the whole api-reference directory. Those links should still work.

File: pkg/apis/admissionregistration/types.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ const (
 // ExternalAdmissionHookConfiguration describes the configuration of initializers.
 type ExternalAdmissionHookConfiguration struct {
 	metav1.TypeMeta
-	// Standard object metadata; More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata.
+	// Standard object metadata; More info: https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata.
```
Comment (thockin) [inline pkg/apis/admissionregistration/types.go]: I've made git.k8s.io/$repo/$path take you to the right place, now, if you want to simplify :)

File: hack/lib/util.sh [hunk 2]
```diff
@@ -202,18 +202,6 @@ kube::util::gen-docs() {
 
   mkdir -p "${dest}/docs/user-guide/kubectl/"
   "${gendocs}" "${dest}/docs/user-guide/kubectl/"
-  mkdir -p "${dest}/docs/admin/"
-  "${genkubedocs}" "${dest}/docs/admin/" "kube-apiserver"
```
Comment (thockin) [inline hack/lib/util.sh]: can we get redi of the definitions of genkubedocs and genfeddocs?  How about those binaries?

**Review by thockin — COMMENTED** [review]
wow!!

File: hack/lib/util.sh (hunk 2 above)
Comment (cblecker) [inline hack/lib/util.sh]: @thockin I actually reverted this change. These docs are autogenerated, and there seems to be some process that I don't know of that replaces these placeholder files with actual generated docs.

To try and split up changes, I didn't want to touch the autogenerated ones *yet*. Just get rid of the static cruft and fix links to go to the right place

File: pkg/apis/admissionregistration/types.go (hunk 1 above)
Comment (cblecker) [inline pkg/apis/admissionregistration/types.go]: fixed

**Comment by cblecker** [comment]
Thinking about it, it may be best to end up splitting this into two PRs.. one to update all the links and other pieces that can be merged by the bot, and a second to remove all this static content (as it would have to be manually merged).

Thoughts?

**Comment by thockin** [comment]
smaller PRs are easier to review...

On Thu, Jun 1, 2017 at 10:44 PM, Christoph Blecker <notifications@github.com

**Comment by cblecker** [comment]
So this PR is now a pure `docs/` deletion PR. Tests won't pass until #46828 is merged into master, as there is one test that unfortunately rely on files in `docs/` existing.

**Comment by cblecker** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by thockin** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by kargakis** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this (ns deletion flake)

**Comment by cblecker** [comment]
/retest
I think the flaky tests have been resolved

**Comment by cblecker** [comment]
@thockin Tests are passing. As this change is prohibited from auto merging, should this be manually merged?

**Comment by thockin** [comment]
@dchen1107 are you ok to manually merge this?

**Comment by thockin** [comment]
@dchen1107 says go.  Fire in the hole!

**Pull request merged**